doc/design/binst-install-command.md

```diff
@@ -0,0 +1,308 @@
+    installCmd.Flags().BoolVar(&debug, "debug", false, "Enable debug logging")
+}
+```
+
+## Implementation Plan
+
+### Phase 1: Core Infrastructure (Foundation)
+1. Create package structure
+2. Implement `pkg/config` using existing code
+3. Add basic CLI command skeleton
+4. Write integration test framework
+
+### Phase 2: Resolution and Download
+1. Implement `pkg/resolve` with OS/arch detection
+2. Implement `pkg/fetch` with retry logic
+3. Add tests for version resolution
+4. Test download functionality
+
+### Phase 3: Verification and Extraction
+1. Implement `pkg/verify` using existing checksum code
+2. Implement `pkg/archive` for tar.gz and zip
+3. Add tests for checksum verification
+4. Test archive extraction
+
+### Phase 4: Installation
+1. Implement `pkg/install` with atomic operations
+2. Complete CLI command implementation
+3. Add dry-run support
+4. Test full installation flow
+
+### Phase 5: Parity Testing
+1. Create parity test suite comparing with generated scripts
+2. Test on multiple platforms (Linux, macOS)
+3. Test with various projects from testdata/
+4. Fix any behavioral differences
+
+## Testing Strategy
+
+### Unit Tests
+- Each package gets comprehensive unit tests
+- Mock external dependencies (GitHub API, filesystem)
+- Test error conditions and edge cases
+
+### Integration Tests
+- Test full installation flow
+- Use testdata configurations
+- Compare results with generated installers
+
+### Parity Tests
+```go
+func TestInstallParity(t *testing.T) {
+    // 1. Generate installer script
+    // 2. Run installer script in container
+    // 3. Run binst install in same container
+    // 4. Compare installed binaries (path, permissions, content)
+}
+```
+
+### Platform Testing
+- Linux: amd64, arm64
+- macOS: amd64, arm64 (including Rosetta 2)
+- Use GitHub Actions matrix builds
+
+## Security Considerations
+
+1. **Checksum Verification**: Always verify checksums before installation
+2. **TLS**: Use HTTPS for all downloads
+3. **Atomic Operations**: Prevent partial installations
+4. **Permission Handling**: Set appropriate file permissions
+5. **Path Validation**: Prevent directory traversal attacks
+
+## Error Handling
+
+Match the shell script's error behavior:
+1. Network errors: Retry with backoff
+2. Checksum mismatch: Fail immediately
+3. Missing assets: Clear error message
+4. Permission errors: Suggest fixes
+
+## Environment Variables
+
+Support the same variables as generated scripts:
+- `GITHUB_TOKEN`: For API authentication
+- `BINSTALLER_BIN`: Override default bin directory
+- Standard proxy variables: `HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY`
+
+## Future Extensibility
+
+This design enables future `binst x` command by:
+1. Reusing resolution/fetch/verify packages
+2. Adding caching layer on top
+3. Executing in temporary directory instead of installing
+
+## Open Questions
+
+1. Should we support parallel asset downloads for faster installation?
```

Comment:
No. We should download only one asset in the first place. 
Exception is checksum file.